### PR TITLE
Sodie pops

### PR DESCRIPTION
**Pull request by dannno**
:cl: Dannno
imageadd: Robust Softdrinks LLC. has sent out new vendies to the stendy.
/:cl:

new vendy machines.
Cola and Snack machine have variants that will be picked at random when the map is loaded. 
That will be added in a separate PR, if this gets merged, because I fucked up editing the map and didn't use map merger.

not an in-game pic but you get it.

I also added some posters for variety's sake and changed how soda cans look.
I also added two new soda brands for fun.

### CODE REVIEW

**Comment by dannno** [comment]
PS. I made juice cartons consistent with other carton items.

**Comment by KorPhaeron** [comment]
no player references sorry

**Comment by dannno** [comment]
wat

**Comment by dannno** [comment]
stfu kor

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -701,6 +701,28 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
...
+	desc = "Uh oh!"
+
+/obj/machinery/vending/snack/random/New()
+    ..()
+    var/T = pick(subtypesof(/obj/machinery/vending/snack) - /obj/machinery/vending/snack/random)
+    new T(get_turf(src))
+    qdel(src)
+
+/obj/machinery/vending/snack/blue
+	icon_state = "snackblue"
```
Comment (Iamgoofball) [inline code/game/machinery/vending.dm]: you could just randomize this on the default snack new so no map edits are required

**Comment by Incoming5643** [comment]
it doesn't look like you added crushed can sprites for the new brands

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -725,12 +747,67 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
...
+
+/obj/machinery/vending/cola/space_up
+	icon_state = "space_up"
+	name = "\improper Space-up! Vendor"
+	desc = "Indulge in an explosion of flavor."
+	product_slogans = "Space-up! Like a hull breach in your mouth."
+
+/obj/machinery/vending/cola/starkist
+	icon_state = "starkist"
+	name = "\improper Star-kist Vendor"
```
Comment (Iamgoofball) [inline code/game/machinery/vending.dm]: same here

File: code/game/machinery/vending.dm (hunk 1 above)
Comment (dannno) [inline code/game/machinery/vending.dm]: Without a snack/random or a cola/random the entire thing doesn't work because of how subtypesof() works.

**Comment by dannno** [comment]
I'm pretty sure I did add crushed versions but I'll double check.

**Comment by dannno** [comment]
as far as I can see all cans have a crushed can

File: code/game/machinery/vending.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/game/machinery/vending.dm]: nono, I mean ditch the random subtype annd make it on the normal soda type to randomize

not exactly OOP but it works

**Comment by Niknakflak** [comment]
Is shambler the reference
The product slogans that are gibberish seem dumb even if they are a reference

**Comment by dannno** [comment]
it's not a reference to anything, it's original

**Comment by Niknakflak** [comment]
dank bruh

**Comment by dannno** [comment]
it's just like a neo eldritch occultist vibe, nah mean dah?

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -725,12 +747,67 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
...
 	products = list(/obj/item/weapon/reagent_containers/food/drinks/soda_cans/cola = 10,/obj/item/weapon/reagent_containers/food/drinks/soda_cans/space_mountain_wind = 10,
 					/obj/item/weapon/reagent_containers/food/drinks/soda_cans/dr_gibb = 10,/obj/item/weapon/reagent_containers/food/drinks/soda_cans/starkist = 10,
-					/obj/item/weapon/reagent_containers/food/drinks/soda_cans/space_up = 10,
+					/obj/item/weapon/reagent_containers/food/drinks/soda_cans/space_up = 10,/obj/item/weapon/reagent_containers/food/drinks/soda_cans/pwr_game = 10,
 					/obj/item/weapon/reagent_containers/food/drinks/soda_cans/lemon_lime = 10)
-	contraband = list(/obj/item/weapon/reagent_containers/food/drinks/soda_cans/thirteenloko = 6)
+	contraband = list(/obj/item/weapon/reagent_containers/food/drinks/soda_cans/thirteenloko = 6,/obj/item/weapon/reagent_containers/food/drinks/soda_cans/shamblers = 6)
 	premium = list(/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/filled/nuka_cola = 1)
 	refill_canister = /obj/item/weapon/vending_refill/cola
```
Comment (Supermichael777) [inline code/game/machinery/vending.dm]: Dont forget to change the restockers to load them up with the extera charges

Comment (dannno) [inline code/game/machinery/vending.dm]: I swear I did, don't know where it went

**Comment by ChangelingRain** [comment]
This changes the arcade sprites was that intentional

File: code/game/machinery/vending.dm (hunk 3 above)
Comment (dannno) [inline code/game/machinery/vending.dm]: YEAH I DID ITS RIGHT IN THE THING YOU GOLF

**Comment by dannno** [comment]
Yes, I forgot to mention that lol my bad
I changed the arcade sprites because they're basically just vending machines kind of? Like they're always around vending machines.

**Comment by Niknakflak** [comment]
well upload a picture

**Comment by Iatots** [comment]
>generic cross with three lines and rot13 swaps
Wow, so obscure, much occult.

**Comment by xxalpha** [comment]
Pretty neat. 📸

**Comment by WJohn** [comment]
These looks really nice.

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 4]
```diff
@@ -364,6 +364,27 @@
 		M.bodytemperature = max(310, M.bodytemperature - (8 * TEMPERATURE_DAMAGE_COEFFICIENT)) //310 is the normal bodytemp. 310.055
 	..()
 
+/datum/reagent/consumable/pwr_game
+	name = "Pwr Game"
+	description = "The only drink with the PWR that true gamers crave."
+	id = "pwr_game"
+	color = "#9385bf" // rgb: 58, 52, 75
```
Comment (coiax) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Gonna need a flavour description for your two new drinks if and when #24323 get merged.

For reference:

Space Cola - > "cola"
Nuka Cola -> "the future"
Space Mountain Wind -> "sweet citrus soda"
Dr. Gibb -> "cherry soda"
Space-Up -> "cherry soda"
Lemon Lime -> "tangy lime and lemon soda"

To be honest, feel free to come up with anything that's appropriate and/or funny.

File: code/game/objects/items/weapons/vending_items.dm [hunk 5]
```diff
@@ -51,7 +51,7 @@
 /obj/item/weapon/vending_refill/cola
 	machine_name = "Robust Softdrinks"
 	icon_state = "refill_cola"
-	charges = list(20, 2, 1)//of 60 standard, 6 contraband, 1 premium
+	charges = list(25, 4, 1)//of 75 standard, 12 contraband, 1 premium
 	init_charges = list(20, 2, 1)
```
Comment (lordpidey) [inline code/game/objects/items/weapons/vending_items.dm]: Do you also need to change init_charges?  I'm not familiar with how these work.

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 4 above)
Comment (RemieRichards) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]: lemon lime should be verbatim what the guy from deus ex says, further adding to the reference.

File: code/game/objects/items/weapons/vending_items.dm (hunk 5 above)
Comment (dannno) [inline code/game/objects/items/weapons/vending_items.dm]: fuckin probably lol

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 4 above)
Comment (XDTM) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Maybe nuka cola should taste like war. It never changed.

**Comment by ChangelingRain** [comment]
rot13 text should probably only be ratvar; maybe try a different rotation?

**Comment by dannno** [comment]
Fair, I will

**Comment by ChangelingRain** [comment]
Don't forget to leave a comment saying which ROT# you need to translate text to/from it.

**Comment by dannno** [comment]
Fckfkcmkf
It's rot17

**Comment by ChangelingRain** [comment]
I mean a code comment!

**Comment by dannno** [comment]
I KNOW WHAT YOU MEAN

**Comment by Nienhaus** [comment]
Nice.

**Comment by Iatots** [comment]
I know it's (probably) rot13 because ss13, but since ratvar is clockpunk wouldn't it make more sense to be 12?

**Comment by Niknakflak** [comment]
These are going to look pretty good when approaching vending machines from the bottom.  But when approaching from the side which is already mapped, its gonna look weird and bad

**Comment by XDTM** [comment]
rot13 is used because another rot13 permutation gives back the original text, ss13 is a coincidence

**Comment by dannno** [comment]
@Niknakflak It's really no different than the current ones
I'd say it's only the fact that i'm touching them at all that brings your attention to it

File: code/game/objects/effects/contraband.dm [hunk 6]
```diff
@@ -54,7 +54,15 @@ list(name = "- Punch Shit", desc = " Fight things for no reason, like a man!"),
...
-list(name = "- Busty Backdoor Xeno Babes 6", desc = " Get a load, or give, of these all natural Xenos!") )
+list(name = "- Busty Backdoor Xeno Babes 6", desc = " Get a load, or give, of these all natural Xenos!"),
+list(name = "- Robust Softdrinks", desc = " Robust Softdrinks: More robust than a toolbox to the head!"),
+list(name = "- Shambler's Juice", desc = "~Shake me up some of that Shambler's Juice!~"),
+list(name = "- Pwr Game", desc = "The POWER that gamers CRAVE! In partnership with Vlad's Salad."),
+list(name = "- Sun-kist", desc = "Drink the stars!"),
+list(name = "- Space Cola", desc = "Your favorite cola, in space."),
+list(name = "- Space-Up!", desc = "Sucked out into space by the FLAVOR!"),
+list(name = "- Kudzu", desc = "A poster advertising a movie about plants. How dangerous could they possibly be?"),
+list(name = "- Masked Men", desc = "A poster advertising a movie about some masked men.") )
```
Comment (optimumtact) [inline code/game/objects/effects/contraband.dm]: MASKETTA MAN

**Comment by Niknakflak** [comment]
I will guess I will have to see them ingame

**Comment by KorPhaeron** [comment]
wow the player reference got merged

**Comment by Iatots** [comment]
Can you spell it out for us?
Is it bumba? Google says latvian for ball, but I really can't make out the sprite.

**Comment by dannno** [comment]
baited

**Comment by dannno** [comment]
he's talking about james kudzu

**Comment by peoplearestrange** [comment]
I know tis already been merged. But honestly, this looks really dumb: 
ss 2017-03-08 at 11 49 25

**Comment by xxalpha** [comment]
So just like the old sprites.

**Comment by Incoming5643** [comment]
I've already explained how old sprites were more direction agnostic than these ones. By committing to a direction it throws out the ambiguity the old sprites took advantage of. The costs of a literal perspective here.

**Comment by Incoming5643** [comment]
newoldvendinglarge
Literal 5 minute hackjob

**Comment by dannno** [comment]
2 weeks late dumbass

**Comment by Incoming5643** [comment]
I didn't want to step on anyone's toes. I obviously don't think 3/4ths is a good idea but it's your prerogative to try if you want.

Never too late though.

**Comment by xxalpha** [comment]
Those aren't direction agnostic either, I think you are just used to them.

**Comment by optimumtact** [comment]
ding ding ding xxalpha nailed it you don't notice the old one cause you're used to it.

**Comment by PKPenguin321** [comment]
Nah, diagonal down perspective is more ambiguous than straight down

**Comment by xxalpha** [comment]
no it isnt

trust me, i have no bias as i havent played this in months

**Comment by Incoming5643** [comment]
The fact is that this is a tile based game. When you have something that doesn't fit neatly into a cardinal it becomes clear that the sprites is more a representation of the object than "How this literally looks in 3D space from this location". 

The diagonal representation shows off this concept clearly because it's obviously not "facing south/north/east/west", if it's anything (and that's debatable) it's facing "up". Meanwhile 3/4ths is very obviously facing south and thus invites itself to be taken literally. As people have brought up endlessly this invites a lot of mental gymnastics for justifying how it looks "better" in a tile based world that was never designed to support it.

**Comment by Iatots** [comment]
@dannno put posters 37 to 42 in the proper category, /contraband/legit.
They advertise legal, on-station products.

**Comment by dannno** [comment]
do it yourself

**Comment by peoplearestrange** [comment]
@optimumtact  Its again, about constancy, every other object has the 3/4th's view that the used to have. Machines, computers, tanks, scrubber machines. Everything, every "3D" object has this view and these drink machines, and now boxes dont fit. So they're incredibly jarring. The art style it self doesn't fit the rest of the style of the game, so you notice it more.

**Pull request merged**